src/calendar/index.njk

```diff
@@ -2,44 +2,121 @@
 title: Calendar
 layout: layouts/base.njk
 ---
-<div class="grid gap-4">
-  <h2>Events this week</h2>
 
+<div class="grid grid-flow-row gap-3">
+  <h2 class="text-xl mt-5 mx-4">EVENTS THIS WEEK</h2>
+
+  {% set datevariable = "null" %}
   {% for event in events | eventThisWeek | eventsSort %}
-    <a
-      href="/calendar/{{ event.startDate | formatDate}}/{{ event.slug }}/"
-      class="block border p-4 hover:border-blue"
-      itemscope itemtype="https://schema.org/Event"
-    >
-      <date class="block text-xs" itemprop="startDate" content="{{ event.startDate }}">
-        {{ event.startDate | formatDate('dddd, D MMMM YYYY, h:mma') }} ({{ event.duration }})
-      </date>
 
-      <p itemprop="name">
-        {{ event.title }}
-      </p>
+    {% if event.startDate != datevariable %}
+      <!-- DATE OF EVENT-->
+      <div class="mx-4 text-lg text-black">
+        {{ event.startDate | formatDate('dddd D MMMM') }}
+      </div>
+      <!-- LINE -->
+      <hr class="mx-4 -mt-1 text-black"/>
+      {% set datevariable = event.startDate %}
+    {% endif %}
+
+    <div class="mx-4 grid grid-flow-row grid-cols-2 gap-3">
+      <!-- BORDER, COLOUR & HREF, URL LINK ??? -->
+      <a
+        href="/calendar/{{ event.startDate | formatDate}}/{{ event.slug }}/"
+        class="no-underline"
+        itemscope itemtype="https://schema.org/Event"
+      >
+
+        <!-- NAME OF EVENT -->
+        <p itemprop="name">
+          {{ event.title }}
+        </p>
+
+        <!-- TIME -->
+        <div class="float-left pr-2">{% svgIcon "clock", "h-3 w-3 text-green" %}</div>
+        <date class="block text-xs" itemprop="startDate" content="{{ event.startDate }}">
+          {{ event.startDate | formatDate('h:mma') }} – {{ event.endDate | formatDate('h:mma') }} ({{ event.duration }})
+        </date>
+
+        <!-- LOCATION -->
+        <div class="float-left pr-2">{% svgIcon "pin", "h-3 w-3 text-orange" %}</div>
+        <p class="text-xs" itemprop="location" itemscope itemtype="https://schema.org/Place">
+          {{ event.location}}
+        </p>
+
+        <!-- MORE INFORMATION -->
+        <p class="text-xs hover:underline">
+          More information
+        </p>
+
+        <!-- EVENT TAGS -->
+        {% if event.data.tags %}
+          <div>
+            {% for tag in event.data.tags %}
+              <span class="rounded bg-orange px-1 mr-1 text-xs text-white inline-block">{{ tag }}</span>
+            {% endfor %}
+          </div>
+        {% endif %}
 
-      <p itemprop="location" itemscope itemtype="https://schema.org/Place">{{ event.location }}</p>
-    </a>
+      </a>
+    </div>
   {% endfor %}
 
-  <hr />
 
-  <h2>Future events</h2>
+
+  <h2 class="text-xl mt-6 mx-4">FUTURE EVENTS</h2>
+  {% set futuredate = "null" %}
```

Comment:
As we a repeating pretty much exactly the same code here, can this whole section (and above too) be extracted into an `include`?

```njk
{% for event in events | eventsFuture(7, 'day') | eventsSort %}
  {% include "blocks/eventListing.njk" %}
{% endfor %}
```

See how we get the footer appearing in the template:
https://github.com/theglobechurch/globe-static/blob/main/src/_includes/layouts/base.njk#L39

Reply:
I know what you need me to do here, but not sure the formatting would work with a different variable?